src/main/java/com/netchill/services/streaming/TrailerService.java

```diff
@@ -0,0 +1,44 @@
+package com.netchill.services.streaming;
+
+import com.netchill.api.moviedb.models.Trailer;
+import com.netchill.api.moviedb.models.TrailerKey;
+import com.netchill.api.moviedb.services.trailer.TrailerApiService;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.List;
+
+@Singleton
+public class TrailerService {
+    private TrailerApiService trailerApiClient;
+    private static final String TRAILER_TYPE = "Trailer";
+
+    @Inject
+    private TrailerService(TrailerApiService trailerApiClient) {
+        this.trailerApiClient = trailerApiClient;
+    }
+
+    public TrailerKey getSerieTrailerById(Long id) {
+        Trailer trailer = trailerApiClient.getSerieTrailerById(id);
+        return this.getTrailer(trailer.getKeyList());
+    }
+
+    public TrailerKey getMovieTrailerById(Long id) {
+        Trailer trailer = trailerApiClient.getMovieTrailerById(id);
+        return this.getTrailer(trailer.getKeyList());
+    }
+
+    private TrailerKey getTrailer(List<TrailerKey> trailerKeys) {
+        if (trailerKeys.isEmpty()) {
+            return new TrailerKey();
+        }
+
+        for (TrailerKey key : trailerKeys) {
+            if (key.getType().equals(TRAILER_TYPE)) {
+                return key;
+            }
+        }
+
+        return trailerKeys.get(0);
```

Comment:
tu pourrais passer par des streams ce sera plus simple et plus moderne.

```suggestion
        return trailerKeys.stream()
            .filter(key -> key.getType().equals(TRAILER_TYPE))
            .findFirst() // <== ça renvoie un optional, vide si aucun n'a été trouvé.
            .orElse(trailerKeys.get(0));
```
cf. https://stackify.com/streams-guide-java-8/
& https://www.youtube.com/watch?v=Q93JsQ8vcwY